src/solver/td3.ml

```diff
@@ -272,21 +272,10 @@ module Base =
       in
       let add_sides y x = HM.replace sides y (VS.add x (try HM.find sides y with Not_found -> VS.empty)) in
 
-      let destabilize_ref: (S.v -> unit) ref = ref (fun _ -> failwith "no destabilize yet") in
-      let destabilize x = !destabilize_ref x in (* must be eta-expanded to use changed destabilize_ref *)
+      let destabilize_ref: (?notify_destabilized:(S.v -> bool -> bool -> unit) -> S.v -> unit) ref = ref (fun ?notify_destabilized _ -> failwith "no destabilize yet") in
+      let destabilize ?(notify_destabilized = (fun _ _ _-> ())) x = !destabilize_ref ~notify_destabilized x in (* must be eta-expanded to use changed destabilize_ref *)
 
-      (* Same as destabilize, but returns true if it destabilized a called var, or a var in vs which was stable. *)
-      let rec destabilize_vs x = (* TODO remove? Only used for side_widen cycle. *)
```

Comment:
Maybe this TODO should be investigated. It would be a lot easier to just remove it than complicate the normal `destabilize` further.

Reply:
On the other hand, these used to be common before 20218e763d0ce5f4e9b0f4b878a1f277cf6764cb separated them. Do you think they were separated without good reason?

Reply:
I'm not sure if the removal is a good option at this point in time though, we're actively investigating further strategies to make widening at globals less common / add narrowing, so removing this seems like something we could do after we have investigated this and published our results.